src/js/port_handler.js

```diff
@@ -120,8 +118,16 @@ PortHandler.initialize = function () {
         }
     }, MDNS_INTERVAL);
 
-    // start listening, check after TIMEOUT_CHECK ms
-    this.check();
+    this.reinitialize();    // just to prevent code redundancy
+};
+
+PortHandler.reinitialize = function () {
+    this.initialPorts = false;
+    clearTimeout(this.check_loop);
+    this.showVirtualMode = ConfigStorage.get('showVirtualMode').showVirtualMode;
+    this.showAllSerialDevices = ConfigStorage.get('showAllSerialDevices').showAllSerialDevices;
+    this.check();   // start listening, check after TIMEOUT_CHECK ms
+    this.portPickerElement.change();
```

Comment:
This should be removed as there is already a mechanism in place for detection

Reply:
This it the point of my PR. The existing mechanism is not sufficient to dynamicall change the ui. I have to restart the configurator for the changes to take place, like on virtual fc or show all serial device switch change. I moved these lines from the PortHandler.initialize functio. On the firmware flasher page this reinitialize functuin should be callled because you don't want to flash a virtual fc... If you know what I want to say. That's why I made this function and to change dinamically if I change a switch on the options tab.

Reply:
Have tested my suggestions here and can toggle virtual mode without any problem.

Let's tackle the not flashing a virtual fc in another PR as we just should remove the ability in another way as for now nothing happens if you press flash. We are also going to move to a new API as we need this to be stable.

Reply:
I don't want to flash the virtual FC. It just select the detected board if you click on the flash firmware tab.

Reply:
Yes I appreciate the last part as having tested your PR (with and without my requested changes).

There was a reason we didn't preselect the board on flasher tab as we had to remove that after a lot of testing with older boards, using DFU boot, legacy flashing, switching between FC's and more.

Reply:
Alright, I removed.

Reply:
Yes please also remove line 130 and 171 as the `[trigger(']change[")]` function applies only to firmware tab anyway.

This PR focus is resolving the need for reboot changing show all serial devices and virtual flight control (as I just moved two lines in the past to improve performance as Port Handler is in a 500ms loop).

Other changes should go into another PR if there is any need to fix it.

Reply:
Are these changes unnecessary or they should go to another PR?

Reply:
How to reproduce the bug?

Reply:
You just open the configurator and you wont see the msp box or the manual port input box.
![image](https://user-images.githubusercontent.com/62965528/192071777-9bdd1f3e-ec8d-4a7c-b974-e19613d144e6.png)
![image](https://user-images.githubusercontent.com/62965528/192071783-7ba32e73-494d-4560-9319-b117a6122700.png)

On the left side.

Reply:
> How to reproduce the bug? Yes needs another BUGFIX PR if needed.

So there is no bugfix and feature PR at the same time. It makes sense, I just fixed it since it's a single line of code.

Reply:
Removed